### PR TITLE
fix: sync price slugs with product slugs in update product

### PR DESCRIPTION
**Pull request by agreea**
## What Does this PR Do?
- make it so that changes to product slugs translate to changes in price slugs, enforcing that price slugs are in sync with product slugs

---
## Summary by cubic
Keep price slugs in sync with product slugs when editing a product. Updates the active default price or any newly inserted price, and improves change detection to avoid false positives.

- **Bug Fixes**
  - On product slug change, update the active default price slug, or set it on newly inserted prices; inactive prices are not touched.
  - Override provided price slug to match the product slug when inserting a new price.
  - Skip syncing for default products and respect slug uniqueness constraints.
  - Export and normalize isPriceChanged to treat undefined/null consistently for accurate comparisons.
  - Added comprehensive tests covering slug sync scenarios and isPriceChanged edge cases.

<sup>Written for commit ccf874b1d1fe791912f6a5d1fbab145bf1214518. Summary will update automatically on new commits.</sup>

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive tests for price-change detection across subscription, single-payment, and usage price types.
  * Added extensive tests covering product-to-price slug synchronization and slug-change scenarios.

* **Bug Fixes**
  * Improved price-change detection with stronger normalization and null/undefined handling to avoid false positives.
  * Enhanced product slug synchronization so price records update appropriately when product slugs change (default products remain protected).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: platform/flowglad-next/src/utils/pricingModel.isPriceChanged.test.ts [hunk 1]
```diff
@@ -0,0 +1,856 @@
...
+        intervalCount: 1,
+      })
+      const newPrice = createSubscriptionPriceInsert({
+        intervalCount: 3, // Changed
+      })
+      const result = isPriceChanged(newPrice, currentPrice)
+      expect(result).toBe(true)
+    })
+
+    it('should return true when usageMeterId changes to different value', () => {
```
[the remark this answers is not in the document]
Comment (joeysabs) [reply platform/flowglad-next/src/utils/pricingModel.isPriceChanged.test.ts]: fax

File: platform/flowglad-next/src/utils/pricingModel.isPriceChanged.test.ts [hunk 2]
```diff
@@ -0,0 +1,856 @@
...
+        name: undefined,
+        unitPrice: 2000, // Immutable field changed
+      })
+      const result = isPriceChanged(newPrice, currentPrice)
+      expect(result).toBe(true)
+    })
+  })
+
+  describe('Null vs Undefined Handling', () => {
+    it('should return false when newPrice.slug is undefined and currentPrice.slug is null', () => {
```
[the remark this answers is not in the document]
Comment (joeysabs) [reply platform/flowglad-next/src/utils/pricingModel.isPriceChanged.test.ts]: Fax. Lines 537 and 548

File: platform/flowglad-next/src/utils/pricingModel.test.ts [hunk 3]
```diff
@@ -3138,3 +3141,654 @@ describe('editProductTransaction - Price Updates', () => {
...
+      expect(updatedProduct?.slug).not.toBe('new-default-slug')
+
+      // Verify price slug was not updated
+      const updatedPrice = await adminTransaction(
+        async ({ transaction }) => {
+          return await selectPriceById(defaultPriceId, transaction)
+        }
+      )
+      // Verify the price slug remains unchanged (or null if it was null)
+      expect(updatedPrice?.slug).toBeDefined()
```
[the remark this answers is not in the document]
Comment (joeysabs) [reply platform/flowglad-next/src/utils/pricingModel.test.ts]: Good call we are not actually verifying the price slug. Should probably be:

`expect(updatedPrice?.slug).not.toBe('new-default-slug')`
